File: python/test_arion.py
import json
from arion import dumps_arion, loads_arion


def roundtrip(obj):
    text = dumps_arion(obj)
    back = loads_arion(text)
    assert back == obj
    return text, back


def test_simple_object():
    data = {"name": "Joachim", "age": 37, "active": True}
    roundtrip(data)


def test_nested():
    data = {
        "name": "Joachim",
        "profile": {"role": "Developer", "location": "Austria"},
        "skills": ["Python", "Audio", "AI"],
    }
    roundtrip(data)


def test_multiline():
    data = {
        "bio": "Line one\nLine two\nLine three"
    }
    text, back = roundtrip(data)
    assert "Line two" in text
    assert back == data


def test_forced_strings():
    data = {
        "flag": "true",
        "number_as_string": "37",
        "null_as_string": "null",
    }
    text, back = roundtrip(data)
    assert back == data


if __name__ == "__main__":
    # run basic tests
    for func in [test_simple_object, test_nested, test_multiline, test_forced_strings]:
        func()
    print("All tests passed.")

File: js/test/test_arion.js
import { loadsArion, dumpsArion } from "../src/arion.js";

function assertEqual(a, b) {
  const ja = JSON.stringify(a);
  const jb = JSON.stringify(b);
  if (ja !== jb) {
    throw new Error("Assertion failed: " + ja + " !== " + jb);
  }
}

function roundtrip(obj) {
  const text = dumpsArion(obj);
  const back = loadsArion(text);
  assertEqual(back, obj);
  return text;
}

function runTests() {
  roundtrip({ name: "Joachim", age: 37, active: true });
  roundtrip({
    name: "Joachim",
    profile: { role: "Developer", location: "Austria" },
    skills: ["Python", "Audio", "AI"],
  });
  roundtrip({
    bio: "Line one\nLine two\nLine three",
  });
  roundtrip({
    flag: "true",
    number_as_string: "37",
    null_as_string: "null",
  });
  console.log("All JS ARION tests passed.");
}

runTests();
